Add metadata

rafaparadela opened the pull request:
This PR closes #429. It adds metadata in home/library views

@dialelo @raulraja Could you review please? Thank you

rafaparadela commented:
Note that I'm creating two new issues for:

- Asking Maureen for a new tagline/description.
- Asking Isra for a better poster image.

raulraja commented on server/app/views/templates/base/head.scala.html:
```diff
@@ -0,0 +1,33 @@
+<title>@title</title>
+<meta name="description" content="@description">
+<meta name="author" content="47 Degrees">
+<meta property="og:image" content="@routes.Assets.at("images/poster.png")"/>
```
Would this result in a relative URL? If so I think we need absolute URLs in all instances.

dialelo commented:
Apart from Raúl's comment LGTM :+1:

raulraja commented on server/app/views/templates/base/head.scala.html:
```diff
@@ -0,0 +1,33 @@
+<meta property="og:image" content="@routes.Assets.at("images/poster.png")"/>
+<meta property="og:title" content="@title"/>
+<meta property="og:site_name" content="@current.configuration.getString("application.name").getOrElse("Scala Exercises")"/>
+<meta property="og:url" content="@current.configuration.getString("application.url").getOrElse("https://www.scala-exercises.org")"/>
```
This is not the URL we want to share. If someone shares a section this URL is always going to be the main site and not the section. We need to use the ReverseController of the given action to bring the current absolute URL here.

rafaparadela commented:
Addressed comments:
- Now the URL of poster image is expressed as absolute path.
- Now metadata `url` is sensible to the current path, the target link is consistently to the shared section.

@raulraja Is good to be merged?

raulraja commented:
LGTM
